### PR TITLE
Drop support for Python < 3.7

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Review by dataflake — APPROVED** [review]

File: src/zope/publisher/tests/test_browserrequest.py [hunk 1]
```diff
@@ -488,52 +480,52 @@ def testFormListTypes(self):
...
-        self.assertEqual(list(request.form[u"a"].keys()), [u"x"])
-        self.assertEqual(request.form[u"a"][u"x"], (u"5", u"6"))
-        self.assertEqual(request.form[u"a"].x, (u"5", u"6"))
-        self.assertEqual(str(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(keys, ["a", "b"])
+        self.assertEqual(request.form["b"], "1")
+        self.assertEqual(list(request.form["a"].keys()), ["x"])
+        self.assertEqual(request.form["a"]["x"], ("5", "6"))
+        self.assertEqual(request.form["a"].x, ("5", "6"))
+        self.assertEqual(str(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:509]:
```suggestion
        self.assertEqual(str(request.form["a"]),
```

File: src/zope/publisher/tests/test_browserrequest.py [hunk 2]
```diff
@@ -488,52 +480,52 @@ def testFormListTypes(self):
...
                          "{x: ('5', '6')}")
-        self.assertEqual(repr(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(repr(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:511]:
```suggestion
        self.assertEqual(repr(request.form["a"]),
```

File: src/zope/publisher/tests/test_browserrequest.py [hunk 3]
```diff
@@ -488,52 +480,52 @@ def testFormListTypes(self):
...
-        self.assertEqual(request.form[u"a"][1].x, u"6")
-        self.assertEqual(str(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(keys, ["a", "b"])
+        self.assertEqual(request.form["b"], "1")
+        self.assertEqual(len(request.form["a"]), 2)
+        self.assertEqual(request.form["a"][0]["x"], "5")
+        self.assertEqual(request.form["a"][0].x, "5")
+        self.assertEqual(request.form["a"][1]["x"], "6")
+        self.assertEqual(request.form["a"][1].x, "6")
+        self.assertEqual(str(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:526]:
```suggestion
        self.assertEqual(str(request.form["a"]),
```

File: src/zope/publisher/tests/test_browserrequest.py [hunk 4]
```diff
@@ -488,52 +480,52 @@ def testFormListTypes(self):
...
                          "[{x: '5'}, {x: '6'}]")
-        self.assertEqual(repr(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(repr(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:528]:
```suggestion
        self.assertEqual(repr(request.form["a"]),
```

File: src/zope/publisher/tests/test_browserrequest.py [hunk 5]
```diff
@@ -542,48 +534,48 @@ def testFormMultipleRecordsTypes(self):
...
-        self.assertEqual(list(request.form[u"a"].keys()), [u"x"])
-        self.assertEqual(request.form[u"a"][u"x"], [u"5", u"6"])
-        self.assertEqual(request.form[u"a"].x, [u"5", u"6"])
-        self.assertEqual(str(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(keys, ["a", "b"])
+        self.assertEqual(request.form["b"], "1")
+        self.assertEqual(list(request.form["a"].keys()), ["x"])
+        self.assertEqual(request.form["a"]["x"], ["5", "6"])
+        self.assertEqual(request.form["a"].x, ["5", "6"])
+        self.assertEqual(str(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:563]:
```suggestion
        self.assertEqual(str(request.form["a"]),
```

File: src/zope/publisher/tests/test_browserrequest.py [hunk 6]
```diff
@@ -542,48 +534,48 @@ def testFormMultipleRecordsTypes(self):
...
                          "{x: ['5', '6']}")
-        self.assertEqual(repr(request.form[u"a"]).replace("u'", "'"),
+        self.assertEqual(repr(request.form["a"]).replace("u'", "'"),
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_browserrequest.py:565]:
```suggestion
        self.assertEqual(repr(request.form["a"]),
```

File: src/zope/publisher/tests/test_http.py [hunk 7]
```diff
@@ -205,17 +198,17 @@ class HTTPTests(unittest.TestCase):
...
+        class Folder:
             """Required docstring for the publisher."""
 
-        class Item(object):
+        class Item:
             """Required docstring for the publisher."""
 
             def __call__(self, a, b):
-                return ("%s, %s" % (repr(a), repr(b))).encode('latin1')
+                return ("{}, {}".format(repr(a), repr(b))).encode('latin1')
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_http.py:211]:
```suggestion
                return ("{!r}, {!r}".format(a, b)).encode('latin1')
```

File: src/zope/publisher/tests/test_xmlrpcrequest.py [hunk 8]
```diff
@@ -71,29 +71,29 @@ class XMLRPCTests(unittest.TestCase):
...
-        class Folder(object):
+        class Folder:
             pass
 
-        class Item(object):
+        class Item:
 
             def __call__(self, a, b):
-                return "%s, %s" % (repr(a), repr(b))
+                return "{}, {}".format(repr(a), repr(b))
```
Comment (cjwatson) [inline src/zope/publisher/tests/test_xmlrpcrequest.py:85]:
```suggestion
                return "{!r}, {!r}".format(a, b)
```

**Review by cjwatson — APPROVED** [review]

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**